lib/tag-field/index.tsx

```diff
@@ -25,10 +26,10 @@ type OwnState = {
 };
 
 type StateProps = {
-  allTags: Map<T.TagHash, T.Tag>;
+  tags: Map<T.TagHash, T.Tag>;
```

Comment:
it doesn't look like we intrinsically need the hash here. would it make sense to return a simpler data structure and make a simpler selector to return `T.TagName[]` instead of the map from tag hash to tag record?

Reply:
The tag's hash is being used as the `key` prop for rendering the tag chips but we could use the tag's name instead, in this case it would be enough with `T.TagName[]`. However since the `noteTags` selector is also calculating if a tag is deleted, I'd keep returning `T.Tag` object because we need it for `note-revisions` component.

One thing that we could change is that instead of returning a `Map`, we could simply return an array of `T.Tag`. You can check this change in this [commit](https://github.com/Automattic/simplenote-electron/pull/2817/commits/3a335b55c6f194886673a230b931d671cddac75f).